### PR TITLE
Auto-expand single section editor to utilize available space

### PR DESCRIPTION
**Pull request by narcolepticinsomniac**

### CODE REVIEW

File: edit/sections-editor-section.js [hunk 1]
```diff
@@ -60,13 +60,32 @@ function createResizeGrip(cm) {
...
       const bounds = sectionEl.getBoundingClientRect();
       if (bounds.top < 0 || bounds.bottom > window.innerHeight) {
         window.scrollBy(0, bounds.top);
       }
     }
   }
+
+  document.onreadystatechange = () => {
+    if (document.readyState !== 'loading') {
+      new MutationObserver((_, observer) => {
```
Comment (eight04) [inline edit/sections-editor-section.js]: Why using mutation observer?

Comment (narcolepticinsomniac) [reply edit/sections-editor-section.js]: Wasn't my first choice, but it was the fastest and most reliable of everything I tried. `DOMContentLoaded` was unreliable from one edit page load to another, sometimes it worked, sometimes it didn't. The only `readyState` that worked was 'complete', and it was noticeably late. 

JS isn't my specialty, so if there's anything else as fast and reliable, feel free to hook it up. It needs to get to the calculation as quick as possible, so the delay isn't any longer than it needs to be. With the MO, it's hardly noticeable.

**Comment by eight04** [comment]
If we just want to force the editor to stay at full height and there is only one editor, then we should be able to implement it with plain CSS.

Otherwise, you can write something similar to the autofocus:
https://github.com/openstyles/stylus/blob/19c71868a09ce24972345c928233eda280bbcaf5/edit/sections-editor.js#L439-L441

Even better, listen to the `refresh` event of the editor and avoid `setTimeout`.

**Comment by narcolepticinsomniac** [comment]
>then we should be able to implement it with plain CSS.

Not really the same though, because the js factors in the number of applies-to blocks and calculates the maximum height from there.

>avoid setTimeout

Yeah, works fine in Chromium without it, but in FF, it seemed to expand before the calculation was finished without it. If `refresh` works and isn't any slower, I suppose we could use it.

**Comment by eight04** [comment]
> Not really the same though, because the js factors in the number of applies-to blocks and calculates the maximum height from there.

Haven't tried, but I think it is possible to fit with the applies-to block using flexbox. Pro:

* The editor will be resized automatically when the height of the applies-to block changes.

Con:

* The user can no longer resize the editor manually by dragging the resize-grip.
* The size of the editor will change when the user adds a new section (adds a new editor).

**Comment by narcolepticinsomniac** [comment]
Pros and cons are dead on. I thought of using flexbox, which would be the most obvious way to go about a pure CSS solution, but the cons outweigh the pros IMO, particularly when we already have a function for exactly this height calculation.

**Comment by eight04** [comment]
There is a FIXME which relates to this problem:
https://github.com/openstyles/stylus/blob/514fa3204f2b32fe4eeedab2983e75a9d9a13028/edit/sections-editor.js#L51-L52

@tophf Do you remember how the logic works? https://github.com/openstyles/stylus/commit/2895ce11e271788df0e4f7314b3b981fde086574

**Comment by tophf** [comment]
I don't remember but it seems pretty straightforward judging by the linked code.

**Comment by tophf** [comment]
If I were to implement it now, though, I would probably try using the CodeMirror's init hook or something like that.

**Pull request closed without merging**